internal/controller/resource_manager.go

```diff
@@ -144,74 +147,107 @@ func getErrFromWatchEvent(event watch.Event) error {
 // watchEvents watches Kubernetes resources. The second part of the of the "list
 // and watch" strategy.
 // Any occurring errors are sent through errChanWatch. The watch stops when ctx is cancelled.
-func (r *ResourceManager) watchEvents(ctx context.Context, apiGroup string, watchOptions metav1.ListOptions, mutationChan chan *pb.KubernetesResourceMutation) error {
-	logger := r.logger.With(zap.String("api_group", apiGroup))
+func (r *ResourceManager) watchEvents(ctx context.Context, resourceVersion string, mutationChan chan *pb.KubernetesResourceMutation) error {
+	logger := r.logger
 
-	objGVR := schema.GroupVersionResource{Group: apiGroup, Version: "v1", Resource: r.resourceName}
+	lastKnownResourceVersion := resourceVersion
 
-	watcher, err := r.dynamicClient.Resource(objGVR).Namespace(metav1.NamespaceAll).Watch(ctx, watchOptions)
-	if err != nil {
-		logger.Error("Error setting up watch on resource", zap.Error(err))
+	mutationCount := 0
 
-		return err
+	for {
+		// Run the watcher loop until a restart is requested or an error occurs
+		updatedRV, restart, err := r.runWatcherUntilRestart(ctx, lastKnownResourceVersion, mutationChan, logger, &mutationCount)
```

Comment:
Spell out `RV`. There's no need to invent new acronyms.
```suggestion
		updatedResourceVersion, restart, err := r.runWatcherUntilRestart(ctx, lastKnownResourceVersion, mutationChan, logger, &mutationCount)
```